static/src/javascripts/projects/commercial/modules/comscore.js

```diff
@@ -0,0 +1,47 @@
+// @flow strict
+import config from 'lib/config';
+
+type comscoreGlobals = { c1: string, c2: string, comscorekw?: string };
+
+const comscoreSrc = 'https://sb.scorecardresearch.com/beacon.js';
+const comscoreC1 = '2';
+const comscoreC2 = '6035250';
+
+const getGlobals = (keywords: string): comscoreGlobals => {
+    const globals: comscoreGlobals = { c1: comscoreC1, c2: comscoreC2 };
+
+    if (keywords !== 'Network Front') {
+        globals.comscorekw = keywords;
+    }
+
+    return globals;
+};
+
+export const init = (): Promise<void> => {
+    if (config.get('switches.comscore', false)) {
+        // eslint-disable-next-line no-underscore-dangle
+        window._comscore = window._comscore || [];
+
+        // eslint-disable-next-line no-underscore-dangle
+        window._comscore.push(getGlobals(config.get('page.keywords', '')));
+
+        const s = document.createElement('script');
+        s.id = 'comscore';
+        s.async = true;
+        s.src = comscoreSrc;
+
+        const el = document.getElementsByTagName('head')[0];
+        if (el) {
+            el.appendChild(s);
```

Comment:
👋 Would it be worth switching this module to be part of [third-party-tags](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/commercial/modules/third-party-tags.js)?

TPT module would handle the script insertion for us, wrapped within fastdom (to eliminate layout thrashing)  giving us less to test. The `async` property could be an optional key:

```

declare type ThirdPartyTag = {
    shouldRun: boolean,
    url: string,
    onLoad?: () => any,
    useImage?: boolean,
    async?: boolean,
};
```

Then we can can conditionally set `script.async = service.async` in third-party-tags module.

Reply:
If we need to do something before script load and therefore cannot use the `onLoad` pattern, would the folowing work:

```js
const shouldRun = config.get('switches.comscore', false);

if (shouldRun) {
    // eslint-disable-next-line no-underscore-dangle
    window._comscore = window._comscore || [];
    // eslint-disable-next-line no-underscore-dangle
    window._comscore.push(getGlobals(config.get('page.keywords', '')));
}
```

Reply:
As of yesterday TPT includes an optional `async` (https://github.com/guardian/frontend/pull/22396/files), had to do for Connatix script which is async

Reply:
Hi Kate,
Great suggestion! I considered it but opted for a different solution. Here's my reasoning:
This PR is the first of two steps. The second step will have comscore running differently based on whether or not it has consent. TPT does not offer this option and I felt reworking it to allow for it + the changes you suggested for this step would be too big of a change to put TPT through for a single script.